addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CommandInjectionPlugin.java

```diff
@@ -134,6 +136,54 @@
         // Used for *nix
         // OS_PAYLOADS.put("\"|\"ld", null);
         // OS_PAYLOADS.put("'|'ld", null);
+
+        // Null Byte Payloads
+
+        // No quote payloads
+        NIX_OS_PAYLOADS.put(";" + NIX_TEST_CMD + NULL_BYTE_CHARACTER, NIX_CTRL_PATTERN);
+        NIX_OS_PAYLOADS.put("&" + NIX_TEST_CMD + NULL_BYTE_CHARACTER, NIX_CTRL_PATTERN);
+
+        WIN_OS_PAYLOADS.put("&" + WIN_TEST_CMD + NULL_BYTE_CHARACTER, WIN_CTRL_PATTERN);
+        WIN_OS_PAYLOADS.put("|" + WIN_TEST_CMD + NULL_BYTE_CHARACTER, WIN_CTRL_PATTERN);
+
+        // Double quote payloads
+        NIX_OS_PAYLOADS.put("\"&" + NIX_TEST_CMD + NULL_BYTE_CHARACTER, NIX_CTRL_PATTERN);
+        NIX_OS_PAYLOADS.put("\";" + NIX_TEST_CMD + NULL_BYTE_CHARACTER, NIX_CTRL_PATTERN);
+
+        WIN_OS_PAYLOADS.put("\"&" + WIN_TEST_CMD + NULL_BYTE_CHARACTER, WIN_CTRL_PATTERN);
+        WIN_OS_PAYLOADS.put("\"|" + WIN_TEST_CMD + NULL_BYTE_CHARACTER, WIN_CTRL_PATTERN);
+
+        // Single quote payloads
+        NIX_OS_PAYLOADS.put("'&" + NIX_TEST_CMD + NULL_BYTE_CHARACTER, NIX_CTRL_PATTERN);
+        NIX_OS_PAYLOADS.put("';" + NIX_TEST_CMD + NULL_BYTE_CHARACTER, NIX_CTRL_PATTERN);
+
+        WIN_OS_PAYLOADS.put("'&" + WIN_TEST_CMD + NULL_BYTE_CHARACTER, WIN_CTRL_PATTERN);
+        WIN_OS_PAYLOADS.put("'|" + WIN_TEST_CMD + NULL_BYTE_CHARACTER, WIN_CTRL_PATTERN);
+
+        // Special payloads
+        NIX_OS_PAYLOADS.put(
+                "||" + NIX_TEST_CMD + NULL_BYTE_CHARACTER,
+                NIX_CTRL_PATTERN); // or control concatenation
+        NIX_OS_PAYLOADS.put(
+                "&&" + NIX_TEST_CMD + NULL_BYTE_CHARACTER,
+                NIX_CTRL_PATTERN); // and control concatenation
+        // FoxPro for running os commands
+        WIN_OS_PAYLOADS.put("run " + WIN_TEST_CMD + NULL_BYTE_CHARACTER, WIN_CTRL_PATTERN);
+
+        // uninitialized variable waf bypass
+        insertedCMD = insertUninitVar(NIX_TEST_CMD);
+        // No quote payloads
+        NIX_OS_PAYLOADS.put("&" + insertedCMD + NULL_BYTE_CHARACTER, NIX_CTRL_PATTERN);
+        NIX_OS_PAYLOADS.put(";" + insertedCMD + NULL_BYTE_CHARACTER, NIX_CTRL_PATTERN);
+        // Double quote payloads
+        NIX_OS_PAYLOADS.put("\"&" + insertedCMD + NULL_BYTE_CHARACTER, NIX_CTRL_PATTERN);
+        NIX_OS_PAYLOADS.put("\";" + insertedCMD + NULL_BYTE_CHARACTER, NIX_CTRL_PATTERN);
+        // Single quote payloads
+        NIX_OS_PAYLOADS.put("'&" + insertedCMD + NULL_BYTE_CHARACTER, NIX_CTRL_PATTERN);
+        NIX_OS_PAYLOADS.put("';" + insertedCMD + NULL_BYTE_CHARACTER, NIX_CTRL_PATTERN);
+        // Special payloads
+        NIX_OS_PAYLOADS.put("||" + insertedCMD + NULL_BYTE_CHARACTER, NIX_CTRL_PATTERN);
+        NIX_OS_PAYLOADS.put("&&" + insertedCMD + NULL_BYTE_CHARACTER, NIX_CTRL_PATTERN);
```

Comment:
you are suggesting to have a placeholder in the existing payloads and in case of all the other strength levels make it empty and in case of insane level , we will run with empty payload as well as by adding null byte to the placeholder ? 

one thing in that case is Even if we can run the null byte payload under high level as number of requests are less that 24, we might not be able to do that.

thanks,
Karan

Reply:
> you are suggesting to have a placeholder in the existing payloads and in case of all the other strength levels make it empty and in case of insane level , we will run with empty payload as well as by adding null byte to the placeholder ?

Yup it can just be handled inside a loop probably, if STR is whatever and payload count less that plan but over the collection size use null byte instead of empty. (I'm reviewing on a mobile device right now so it isn't practical to dig into the code, but it seems doable.)

Reply:
yes yes, it is doable. I just wanted to clarify. will do the changes.
thanks,
Karan

Reply:
Hi @kingthorin 

I had started the changes (as you suggested) but following are my thoughts - 

1) Adding placeholders to each payload might not be suitable if we want to add a payload which only suits command injection not null byte injection. (Currently we have some examples like "| cat /etc/passwd#" or "| cat /etc/passwd" or "| cat /etc/passwd |" mean same if we add null byte payload to all of them)

2) if we add it to specific payloads then we need to check if placeholder is present or not, if it is there then only replace and also in case of HIGH Strength, if payload count is less than recommended size for high strength we need to reiterate the payloads (after once iterating by replacing placeholder with empty string), replace placeholder again and then run them until it is equal to recommended size.

I think it is good to have a separate list of payloads which are just associated to null bytes.
please share your thoughts.

thanks,
Karan